utils/stm.stanza

```diff
@@ -0,0 +1,302 @@
+              ;Example map: bund.p => required-pin.q
+              ;Read the name of the bundle pin (ex. "p")
+              val bundle-pin = match(mapping["bundle-pin"]) :
+                (bundle-pin:String) : bundle-pin
+                (bundle-pin) : fatal("Invalid Mapping Bundle Pin")
+              ;Read the name of the required pin (ex. "required-pin")
+              val require = match(mapping["require"]) :
+                (require:String|False) : require
+                (require) : fatal("Invalid Mapping Require")
+              ;Read the name of the mapped pin (ex. "q")
+              val pin = match(mapping["pin"]) :
+                (pin:String) : pin
+                (pin) : fatal("Invalid Mapping Pin")
+              STMSupportMapping(bundle-pin, require, pin)
+          (mappings) : fatal("Invalid Support Mappings")
+        ;Bring the bundle and mappings together to create one STMSupports
+        STMSupports(supported-bundle, mappings)
+    (supports) : fatal("Invalid Supports")
+
+public defn create-bundles (json1:JSON) -> Tuple<STMBundle> :
+  val bundles = Vector<STMBundle>()
+  val json = json1 as JObject
+
+  match(json["bundles"]) :
+    (json:Tuple<JSON>) :
+      ;Each name should be a String bundle name
+      for name in json do :
+        match(name:String) : add(bundles, STMBundle(name))
+        else : fatal("Invalid JSON")
+    (json) :
+      fatal("Invalid JSON")
+  
+  to-tuple(bundles)
+
+;This function creates a Tuple of PinProperties from the bundles field in the JSON
+public defn create-pin-properties (json1:JSON) -> STMPinProperties :
+  val json = json1 as JObject
+  val pin-prop = json["pin-properties"] as JObject
+
+  val generic-pin1 = pin-prop["generic-pin"]
+  var stm-gen-pin
+  match(generic-pin1) :
+    (gen-pin: JObject) :
+      val name = match(gen-pin["name"]) :
+        (name : String) : name
+        (name) : fatal("Invalid Generic Pin Name")
+      val rated-esd = match(gen-pin["rated-esd"]) :
+        (rated-esd: Double) : rated-esd
+        (rated-esd) : fatal("Invalid rated-esd")
+      var max-voltage-data
+      match(gen-pin["max-voltage"]) :
+        (max-volt-data: JObject) :
+          val min-volts = match(max-volt-data["min"]) :
+            (min-volts-double:Double) : min-volts-double
+            (min-volts) : fatal("Invalid max-voltage data")
+          val max-volts = match(max-volt-data["max"]) :
+            (max-volts-double:Double) : max-volts-double
+            (max-volts) : fatal("Invalid max-voltage data")
+          val nominal = match(max-volt-data["nominal"]) :
+            (nominal:True) : nominal
+            (nominal:False) : nominal
+            (nominal) : fatal("Invalid nominal data")
+          max-voltage-data = STMVoltageLimits(min-volts, max-volts, nominal)
+        (max-volt-data) :
+          fatal("Invalid JSON") 
+      stm-gen-pin = STMGenericPin(name, max-voltage-data, rated-esd)
+
+  val power-pin1 = pin-prop["power-pin"]
```

Comment:
Why a `1`?

Reply:
I believe that was used to at one point to distinguish from the `STMPinPoperties`'s `power-pin` field.